parsl/executors/high_throughput/process_worker_pool.py

```diff
@@ -731,9 +731,19 @@ def worker(
         os.sched_setaffinity(0, my_cores)  # type: ignore[attr-defined, unused-ignore]
         logger.info("Set worker CPU affinity to {}".format(my_cores))
 
+    # If CUDA devices, find total number of devices to allow for MPS
+    # See: https://developer.nvidia.com/system-management-interface
+    try:
+        mod_factor = int(os.popen("nvidia-smi -L | wc -l").read())
+    except:
+        mod_factor = None
+        
     # If desired, pin to accelerator
     if accelerator is not None:
-        os.environ["CUDA_VISIBLE_DEVICES"] = accelerator
+        try:
+            os.environ["CUDA_VISIBLE_DEVICES"] = str(int(accelerator)%mod_factor) # multiple workers will share a GPU
```

Comment:
This will order accelerators as [0, 1, 0, 1], which doesn't line up with the memory locality on Polaris (adjacent blocks of cores map to the same GPU). Could we do `int(accelerator) // mod_factor` instead so that it goes [0, 0, 1, 1]? 

Also (pendanty warning). what about naming `mod_factor` to "num_cuda_devices" instead for better clarity?

Reply:
I agree this should be the correct order.  To accomplish it, I think what's needed is something like:
```
procs_per_cuda_device = pool_size//mod_factor
os.environ["CUDA_VISIBLE_DEVICES"] = int(accelerator)//procs_per_cuda_device
```

Reply:
Good catch!

Reply:
this is polaris-specific, right, so we might expect in future users to want a different layout, in the same way that there are different worker<->core pins, right?

Reply:
The previous behavior put the GPU ids in ascending order on every machine, this change aligns with that behavior, so I'm not sure I see how this is Polaris specific.  Users still have the ability to tailor the CPU affinity to the GPU affinity however they choose.  I could imagine applications where you want certain ranks on certain GPUS for mpi communication, but that doesn't apply here?

However, I should modify the documentation to explain the gpus will be assigned in this fashion.